functions.php

```diff
@@ -687,19 +687,62 @@ function get_gmucf_email_options_feed_values() {
  * @return array
  * @author Chris Conover
  **/
-function get_announcement_details() {
+function get_announcement_details( $announcement_ids=array() ) {
 	$announcements = array();
 
-	$response = wp_remote_get( ANNOUNCEMENTS_JSON_URL );
+	/**
+	 * First check to see if there are specific announcement_ids to pull.
+	 * If there are, we need to get the data from each announcement.
+	 */
 
-	if( is_array( $response ) ) {
+	// Slice up the default URL to remove query params and trailing slash
+	$base_url       = preg_replace( "/\/\?.*/", "", ANNOUNCEMENTS_JSON_URL );
+	$front_base_url = preg_replace( "/^(.*)(.*\/api).*$/", "$1", ANNOUNCEMENTS_JSON_URL );
+
+	if ( ! empty( $announcement_ids ) ) {
+		foreach( $announcement_ids as $announcement_id ) {
+			$response      = wp_remote_get( "$base_url/$announcement_id/", array( 'timeout' => 5 ) );
+			$response_code = wp_remote_retrieve_response_code( $response );
+
+			// Continue loop if error code is returned
+			if ( $response_code > 400 ) continue;
+
+			$item = json_decode( wp_remote_retrieve_body( $response ) );
+
+			array_push(
+				$announcements,
+				array(
+					'title' => sanitize_for_email( $item->title ),
+					'permalink' => "$front_base_url/$item->slug"
+				)
+			);
+		}
+
+		/**
+		 * If we have at least one announcement in the array after looping
+		 * through, then we're done and can return out.
+		 */
+		if ( count( $announcements ) > 0 ) {
+			return array_slice( $announcements, 0, 3 );
```

Comment:
Should we have a hard cutoff of 3 announcements? I could see a scenario where we get an email in the future about how there are four announcements selected on the Today GMUCF Options page and only 3 are showing...

If we are restricting it to three, we should adjust that instruction text in the Today Theme above the announcements field to say "Select three announcements":
https://github.com/UCF/Today-Bootstrap/blob/rc-v2.10.0/dev/acf-export.json#L444

Reply:
I was going with the current cutoff we have in place with the automatically retrieved announcements, but yeah we can remove the restriction when they're being explicitly defined.